src/js/inPageNav.js

```diff
@@ -4,54 +4,160 @@ class InPageNav {
+		if (this.opts.activeNavItemClass) {
+			this.activeClass = this.opts.activeNavItemClass;
+		} else {
+			this.activeClass = 'o-in-page-nav-item--active';
+		}
+
+		// The root of the selector for matching a page section to a menu nav item
+		if (this.opts.navItemClassRoot) {
+			this.navItemSelectorRoot = this.opts.navItemClassRoot;
+		} else {
+			this.navItemSelectorRoot = '.o-in-page-nav__item--';
+		}
+
+		this.headings = InPageNav.calculateHeadings(this.headingsSelector, this.containerEl);
+
+		/* add inner wrapper div to prevent the margins collapsing */
+		let inner = document.createElement('div');
+		inner.style.marginTop = '-1px';
+		inner.style.paddingTop = '1px';
+		inner.innerHTML = this.inPageNavEl.innerHTML;
+		this.inPageNavEl.innerHTML = '';
+		this.inPageNavEl.appendChild(inner);
 
 		this.activeMenuItem;
 
+		this.dockPoint = InPageNav.offset(this.inPageNavEl); // don't recalc this unless the viewport changes.
+
 		// Add viewport listeners
 		Viewport.listenTo('scroll');
 		Viewport.listenTo('resize');
 
 		this.scrollWindowHandler = this.scrollWindowHandler.bind(this);
-		//this.showHideMenu = this.showHideMenu.bind(this);
+		this.showHideMenu = this.resizeWindowHandler.bind(this);
 
 		document.addEventListener('oViewport.scroll', this.scrollWindowHandler);
-		//document.addEventListener('oViewport.resize', this.showHideMenu);
+		document.addEventListener('oViewport.resize', this.resizeWindowHandler);
+	}
+
+	static getOptions(inPageNavEl){
+		let opts = {};
+		if (inPageNavEl.hasAttribute('data-o-in-page-nav-headings-selector')) {
+			opts.headingsSelector = inPageNavEl.getAttribute('data-o-in-page-nav-headings-selector');
+		}
+		if (inPageNavEl.hasAttribute('data-o-in-page-nav-headings-container-el')) {
+			opts.containerEl = inPageNavEl.getAttribute('data-o-in-page-nav-headings-container-el');
+		}
+		if (inPageNavEl.hasAttribute('data-o-in-page-nav-active-nav-item-class')) {
+			opts.activeNavItemClass = inPageNavEl.getAttribute('data-o-in-page-nav-active-nav-item-class');
+		}
+		if (inPageNavEl.hasAttribute('data-o-in-page-nav-nav-item-class-root')) {
+			opts.navItemClassRoot = inPageNavEl.getAttribute('data-o-in-page-nav-nav-item-class-root');
+		}
+
+		return opts;
 	}
-/*
-	showHideMenu(){}
 
-	showMenu(){
-		this.inPageNavEl.setAttribute('aria-hidden', false);
+	resizeWindowHandler(){
+		/* TODO: show hide the menu according to the window width */
+
+		/* the dock point may have changed if the page has reflowed, so recalc that */
+		this.dockPoint = InPageNav.offset(this.inPageNavEl);
+
+		/* the headings may have changed position so recalc those */
+		this.headings = InPageNav.calculateHeadings();
+
 	}
-	hideMenu(){
-		this.inPageNavEl.setAttribute('aria-hidden', true);
+
+
+	// If the viewport resizes vertically, (say, because a section is open/closed) these need to be recaculated
+	static calculateHeadings(selector, rootEl) {
+		const elSelector = selector || 'h2';
```

Comment:
these values are hard-coded a couple of times, would it be worth storing them as defaults in the class?